operator.md

```diff
@@ -0,0 +1,120 @@
+# Deploy kfctl as an Operator
+
+## Deployment Instructions
+1. Clone this repository and deploy the CRD and controller
+```shell
+# git clone https://github.com/kubeflow/kfctl.git && cd kfctl
+kubectl create ns operators
+kubectl create -f deploy/crds/kfdef.apps.kubeflow.org_kfdefs_crd.yaml
+kubectl create -f deploy/service_account.yaml
+kubectl create -f deploy/cluster_role_binding.yaml
+kubectl create -f deploy/operator.yaml
+```
+
+2. Deploy KfDef. You can optionally apply ResourceQuota if your Kubernetes version is 1.15+, which will allow only one _kfdef_ instance or one deployment of Kubeflow on this cluster, which follows the singleton model.
+we use ResourceQuota to provide constraints that only one instance of kfdef is allowed within the Kubeflow namespace.
+```shell
+kubectl create ns kubeflow
+# kubectl create -f deploy/crds/kfdef_quota.yaml # only deploy this if the k8s cluster is 1.15+ and has resource quota support
+kubectl create -f <kfdef> -n kubeflow
+```
+
+## Testing Watcher and Reconciler
+One of the major benefits of using kfctl as an Operator is to leverage the functionalities around being able to watch and reconcile your Kubeflow deployments. The Operator is watching all the resources with the `kfctl` label. If one of the resources is deleted, 
+the reconciler will be triggered and re-apply the kfdef to the Kubernetes Cluster.
+
+1. Check the tf-job-operator deployment is running.
+```shell
+kubectl get deploy -n kubeflow tf-job-operator
+# NAME                                          READY   UP-TO-DATE   AVAILABLE   AGE
+# tf-job-operator                               1/1     1            1           7m15s
+```
+
+2. Delete the tf-job-operator deployment
+```shell
+kubectl delete deploy -n kubeflow tf-job-operator
+# deployment.extensions "tf-job-operator" deleted
+```
+
+3. Wait for 10 to 15 seconds, then check the tf-job-operator deployment again. 
+You will be able to see that the deployment is being recreated by the Operator's reconciliation logic
+ 
+```Shell
+kubectl get deploy -n kubeflow tf-job-operator
+# NAME                                          READY   UP-TO-DATE   AVAILABLE   AGE
+# tf-job-operator                               0/1     0            0           10s
+```
+
+## Delete KubeFlow
+Delete KubeFlow deployment
+```shell
+kubectl delete kfdef -n kubeflow --all
+```
+
+Delete KubeFlow Operator
+```shell
+kubectl delete -f deploy/operator.yaml
+kubectl delete -f deploy/cluster_role_binding.yaml
+kubectl delete -f deploy/service_account.yaml
+kubectl delete -f deploy/crds/kfdef.apps.kubeflow.org_kfdefs_crd.yaml
+kubectl delete ns operators
+```
+
+## TroubleShooting
+- When deleting the KubeFlow deployment, it's using kfctl delete in the background where it only deletes the deployment namespace. 
+This will make some of Kuebflow pod deployments hanging because _mutatingwebhookconfigurations_ are cluster-wide 
```

Comment:
`s/Kuebflow/Kubeflow/`

Also I don't this you want to suggest to deploy **all** webhooks as that might influence other applications using webhooks